Add DDNS API cdmon.org

Serrallonga25 opened the pull request:
Added support for DDNS API https://dinamico.cdmon.org

SuperSandro2000 commented on ddclient.in:
```diff
@@ -5450,6 +5460,153 @@ sub nic_1984_update {
+
+Example ${program}.conf file entries:
+  ## single host update
+  protocol=cdmon,                                               \\
+  login=my-my-cdmon.org-login,                                     \\
+  password=my-cdmon.org-password                                   \\
```
```suggestion
  protocol=cdmon,                                               \\
  login=my-my-cdmon.org-login,                                  \\
  password=my-cdmon.org-password                                \\
```

SuperSandro2000 commented on ddclient.in:
```diff
@@ -5450,6 +5460,153 @@ sub nic_1984_update {
+  protocol=cdmon,                                               \\
+  login=my-my-cdmon.org-login,                                     \\
+  password=my-cdmon.org-password                                   \\
+  myhost.mydomain.cat
```
```suggestion
  domain.example.com
```

SuperSandro2000 commented on ddclient.in:
```diff
@@ -5450,6 +5460,153 @@ sub nic_1984_update {
+##
+######################################################################
+sub dinamico_cdmon_update {
+
+
+    debug("\ndinamico_cdmon_update -------------------");
```
```suggestion
    debug("\ndinamico_cdmon_update -------------------");
```

SuperSandro2000 commented on ddclient.in:
```diff
@@ -5450,6 +5460,153 @@ sub nic_1984_update {
+    ## update each configured host
+    foreach my $h (@_) {
+        my $ip = delete $config{$h}{'wantip'};
+        info("setting IP address to %s for %s", $ip, $h);
+        verbose("UPDATE:","updating %s", $h);
```
This is duplicated

Serrallonga25 replied:
I left it that way because I saw that both “sub nic_changeip_update {” line 5656 and “sub nic_1984_update {” line 5426 have it

SuperSandro2000 commented on ddclient.in:
```diff
@@ -5450,6 +5460,153 @@ sub nic_1984_update {
+        info("setting IP address to %s for %s", $ip, $h);
+        verbose("UPDATE:","updating %s", $h);
+
+        my $url;
+        $url   = "https://dinamico.cdmon.org/onlineService.php";
+        $url  .= "?enctype=MD5";
+        $url  .= "&n=$config{$h}{'login'}";
+        $url  .= "&p=$config{$h}{'password'}";
+        $url  .= "&cip=$ip" if $ip;
```
```suggestion
        my $url = "https://dinamico.cdmon.org/onlineService.php?enctype=MD5&n=$config{$h}{'login'}&p=$config{$h}{'password'}";
        $url  .= "&cip=$ip" if $ip;
```

Can we use something other than MD5? MD5 is insecure for cryptography.

Serrallonga25 replied:
It is the only thing that the provider accepts, the password must be in MD5

SuperSandro2000 commented on ddclient.in:
```diff
@@ -5450,6 +5460,153 @@ sub nic_1984_update {
+                if (exists $params->{'newip'})    {$newip = $params->{'newip'}; $msg  .= sprintf " :IP address set to newip:%s", $newip;}
+                if (exists $params->{'temps'})    {$msg  .= sprintf ", the changes will not be operational until after %d minutes", int(int($params->{'temps'})/60000);}
+
+                #$config{$h}{'ip'}     = $ip;
```
```suggestion
```

SuperSandro2000 commented on ddclient.in:
```diff
@@ -5450,6 +5460,153 @@ sub nic_1984_update {
+                    $msg  .= sprintf " : sent IP address %s", $ip;
+                }
+
+                if (exists $params->{'newip'})    {$newip = $params->{'newip'}; $msg  .= sprintf " :IP address set to newip:%s", $newip;}
+                if (exists $params->{'temps'})    {$msg  .= sprintf ", the changes will not be operational until after %d minutes", int(int($params->{'temps'})/60000);}
```
```suggestion
                if (exists $params->{'newip'}) {
                  $newip = $params->{'newip'};
                  $msg  .= sprintf " :IP address set to newip:%s", $newip;
                }
                if (exists $params->{'temps'}) {
                  $msg  .= sprintf ", the changes will not be operational until after %d minutes", int(int($params->{'temps'})/60000);
                }
```

Serrallonga25 commented:
Hello I have added and modified some changes in the descriptive text
When downloading and compiling the code I get an error that did not come out before.

WARNING:  skipping host: my.domain.cat: 'server=1' is an invalid fully qualified host name and optional port number.

SuperSandro2000 commented:
There is another merge conflict. I cannot help you with the error. Please let me know when you fixed it.

Serrallonga25 commented:
> There is another merge conflict. I cannot help you with the error. Please let me know when you fixed it.
Hello SuperSandro, the conflict and the error have already been resolved
